### PR TITLE
SolGov Representative

### PR DESCRIPTION
**Pull request by MarkSuckerberg**
## About The Pull Request
Adds the SolGov Rep to replace both the lawyers and the seldom-used lieutenant (sorry ogan, I know you just made envirosuits for them but I'm sure we can find a good use for them). Neither jobs do much, so I thought combining the two and giving them an actual responsibility would be a nice change of pace.

TODO:
- [X] - ~~Map in SGV Offices~~ Maybe in a different PR
- [x] - Finish replacing lieutenant gear
- [x] - ~~Possibly make it so the second spawned rep is a subordinate of the first?~~ nah
- [x] - WIKI PAAAAGE ([done](https://wiki.boomerstation.space/Government_Attach%C3%A9), full thanks to windowserrors
- [x] - I just realized I forgot to give them a secHUD icon, do that

## Why It's Good For The Game
See above

## Changelog
:cl:
add: A SolGov representative has been stationed on the station in order to make sure it's following space law and is up to code.
add: A SolGov radio channel has been added for SolGov ERTs as well as the one or two representatives
tweak: The SGR has a letter opener (knife with alternate skins like the det revolver) that does 15 damage instead of the lieutenant's gun.
rscadd: Added a new ID card variant for the SGR
rscadd: Resprited briefcases and the leather satchel to make the SGR's loadout a bit less blinding.
del: The lawyer(s) is dead.
del: The lieutenant is dead, if anyone even remembered them.
/:cl:

### CODE REVIEW

**Comment by dinnerwoerror** [comment]
dude, it's been 3 months since you made the pull request for the department holograms, i'm not exactly shedding any tears at the sunken cost

**Comment by dinnerwoerror** [comment]
[k done](https://wiki.boomerstation.space/Government_Attach%C3%A9)

**Comment by dinnerwoerror** [comment]
there i kind of cleaned up the wiki page

**Comment by dinnerwoerror** [comment]
also, since the lieutenant's revolver is being relegated to admin spawn only, shouldn't we like, bundle it into the advanced energy gun research node or something like that so it doesn't get put in the big warehouse of code where the US government probably put the ark of the covenant

**Comment by triplezeta** [comment]
you know, we've already given sec hybrid tasers, what's the worst that could happen

**Review by triplezeta — APPROVED** [review]
New sprites are excellent and resprites of things are a huge improvement over the old! Love it!

**Comment by Baldir-Odensen** [comment]
Now quickly add black outlines

**Comment by Maldaris** [comment]
Can you verify that this role is exempted from antag a-la heads? If not, and there is no intention to do so, @dinnerwoerror needs to change the last section of the wiki page

**Comment by Maldaris** [comment]
So it looks like SG Rep is exempted from everything *but* traitor. Is that intentional?

**Comment by dinnerwoerror** [comment]
well if you look at the jobs code you can see that the guy spawns with a mindshield so I assumed as much

**Comment by Maldaris** [comment]
image
Minor annoyance: capitalization

File: code/controllers/subsystem/blackbox.dm [hunk 1]
```diff
@@ -148,6 +148,8 @@ SUBSYSTEM_DEF(blackbox)
 			record_feedback("tally", "radio_usage", 1, "supply")
 		if(FREQ_CENTCOM)
 			record_feedback("tally", "radio_usage", 1, "centcom")
+		if(FREQ_SOLGOV)
+			record_feedback("tally", "radio_usage", 1, "solgov")
```
Comment (Superyodeler) [inline code/controllers/subsystem/blackbox.dm:152]:
```suggestion
		if(FREQ_SOLGOV)		// Wasp Edit - SolGov Rep
			record_feedback("tally", "radio_usage", 1, "solgov")		// Wasp Edit - SolGov Rep
```
This or move to modular folder

File: code/__DEFINES/access.dm [hunk 2]
```diff
@@ -85,8 +85,8 @@
 
 /// Cloning room and clone pod ejection
 #define ACCESS_CLONING 68
-/// Lieutenant's Office
-#define ACCESS_LIEUTENANT 90 //Let's give em plenty of room, now
+/// SGR Office
+#define ACCESS_SOLGOV 90 //Let's give em plenty of room, now
```
Comment (Superyodeler) [inline code/__DEFINES/access.dm:89]: Could probably go in wasp_defines, otherwise margularize

File: code/__DEFINES/radio.dm [hunk 3]
```diff
@@ -46,6 +46,10 @@
 #define RADIO_KEY_CENTCOM "y"
 #define RADIO_TOKEN_CENTCOM ":y"
 
+#define RADIO_CHANNEL_SOLGOV "SolGov"
+#define RADIO_KEY_SOLGOV "z"
+#define RADIO_TOKEN_SOLGOV ":z"
+
```
Comment (Superyodeler) [inline code/__DEFINES/radio.dm:52]: Move to modular folder

File: code/__DEFINES/radio.dm [hunk 4]
```diff
@@ -57,6 +61,7 @@
 #define FREQ_CTF_RED 1215  // CTF red team comms frequency, red
 #define FREQ_CTF_BLUE 1217  // CTF blue team comms frequency, blue
 #define FREQ_CENTCOM 1337  // CentCom comms frequency, gray
+#define FREQ_SOLGOV 1345 // SolGov comms frequency, dark blue
```
Comment (Superyodeler) [inline code/__DEFINES/radio.dm:64]: Modularize

File: code/game/area/Space_Station_13_areas.dm [hunk 5]
```diff
@@ -314,7 +314,7 @@ NOTE: there are two lists of areas in the end of this file: centcom and station
 	lighting_brightness_tube = 6
 
 /area/lieutenant
-	name = "Lieutenant's Office"
+	name = "SolGov Office"
```
Comment (Superyodeler) [inline code/game/area/Space_Station_13_areas.dm:317]:
```suggestion
	name = "SolGov Office"		// Wasp Edit - SolGov Rep
```
This or move to modular folder.

File: code/game/communications.dm [hunk 6]
```diff
@@ -98,6 +98,7 @@ GLOBAL_LIST_INIT(radiochannels, list(
 	RADIO_CHANNEL_ENGINEERING = FREQ_ENGINEERING,
 	RADIO_CHANNEL_SECURITY = FREQ_SECURITY,
 	RADIO_CHANNEL_CENTCOM = FREQ_CENTCOM,
+	RADIO_CHANNEL_SOLGOV = FREQ_SOLGOV,
```
Comment (Superyodeler) [inline code/game/communications.dm:101]:
```suggestion
	RADIO_CHANNEL_SOLGOV = FREQ_SOLGOV,		// Wasp Edit - SolGov Rep
```

File: code/game/communications.dm [hunk 7]
```diff
@@ -114,6 +115,7 @@ GLOBAL_LIST_INIT(reverseradiochannels, list(
 	"[FREQ_ENGINEERING]" = RADIO_CHANNEL_ENGINEERING,
 	"[FREQ_SECURITY]" = RADIO_CHANNEL_SECURITY,
 	"[FREQ_CENTCOM]" = RADIO_CHANNEL_CENTCOM,
+	"[FREQ_SOLGOV]" = RADIO_CHANNEL_SOLGOV,
```
Comment (Superyodeler) [inline code/game/communications.dm:118]:
```suggestion
	"[FREQ_SOLGOV]" = RADIO_CHANNEL_SOLGOV,		// Wasp Edit - SolGov Rep
```

File: code/game/gamemodes/changeling/changeling.dm [hunk 8]
```diff
@@ -12,7 +12,7 @@ GLOBAL_VAR(changeling_team_objective_type) //If this is not null, we hand our th
 	antag_flag = ROLE_CHANGELING
 	false_report_weight = 10
 	restricted_jobs = list("AI", "Cyborg")
-	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician", "Lieutenant", "Prisoner") // Waspstation edit - Brig Physicians, Second Officer
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician", "SolGov Representative", "Prisoner") // Waspstation edit - Brig Physicians, Second Officer
```
Comment (Superyodeler) [inline code/game/gamemodes/changeling/changeling.dm:15]:
```suggestion
	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician", "SolGov Representative", "Prisoner") // Waspstation edit - Brig Physicians, Solgov Rep
```

File: code/game/gamemodes/cult/cult.dm [hunk 9]
```diff
@@ -36,7 +36,7 @@
 	report_type = "cult"
 	antag_flag = ROLE_CULTIST
 	false_report_weight = 10
-	restricted_jobs = list("Chaplain","AI", "Cyborg", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician", "Lieutenant", "Prisoner") // Waspstation edit - Brig Physicians, Second Officer
+	restricted_jobs = list("Chaplain","AI", "Cyborg", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician", "SolGov Representative", "Prisoner") // Waspstation edit - Brig Physicians, Second Officer
```
Comment (Superyodeler) [inline code/game/gamemodes/cult/cult.dm:39]:
```suggestion
	restricted_jobs = list("Chaplain","AI", "Cyborg", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician", "SolGov Representative", "Prisoner") // Waspstation edit - Brig Physicians, Solgov Rep
```

File: code/game/gamemodes/revolution/revolution.dm [hunk 10]
```diff
@@ -13,7 +13,7 @@
 	report_type = "revolution"
 	antag_flag = ROLE_REV
 	false_report_weight = 10
-	restricted_jobs = list("Security Officer", "Warden", "Detective", "AI", "Cyborg","Captain", "Head of Personnel", "Head of Security", "Chief Engineer", "Research Director", "Chief Medical Officer", "Brig Physician", "Lieutenant", "Prisoner") // Waspstation edit - Brig Physicians, Second Officer
+	restricted_jobs = list("Security Officer", "Warden", "Detective", "AI", "Cyborg","Captain", "Head of Personnel", "Head of Security", "Chief Engineer", "Research Director", "Chief Medical Officer", "Brig Physician", "SolGov Representative", "Prisoner") // Waspstation edit - Brig Physicians, Second Officer
```
Comment (Superyodeler) [inline code/game/gamemodes/revolution/revolution.dm:16]:
```suggestion
	restricted_jobs = list("Security Officer", "Warden", "Detective", "AI", "Cyborg","Captain", "Head of Personnel", "Head of Security", "Chief Engineer", "Research Director", "Chief Medical Officer", "Brig Physician", "SolGov Representative", "Prisoner") // Waspstation edit - Brig Physicians, Solgov Rep
```

File: code/game/gamemodes/traitor/traitor.dm [hunk 11]
```diff
@@ -12,7 +12,7 @@
 	antag_flag = ROLE_TRAITOR
 	false_report_weight = 20 //Reports of traitors are pretty common.
 	restricted_jobs = list("Cyborg")//They are part of the AI if he is traitor so are they, they use to get double chances
-	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician", "Lieutenant", "Prisoner")		// Waspstation edit - Brig Physicians, Second Officer
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician", "Prisoner")		// Waspstation edit - Brig Physicians, Second Officer
```
Comment (Superyodeler) [inline code/game/gamemodes/traitor/traitor.dm:15]:
```suggestion
	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Brig Physician", "Prisoner")		// Waspstation edit - Brig Physicians, Solgov Rep
```

File: code/game/machinery/computer/card.dm [hunk 12]
```diff
@@ -37,7 +37,7 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 		"Research Director",
 		"Chief Medical Officer",
 		"Brig Physician",
-		"Lieutenant",
+		"SolGov Representative",
```
Comment (Superyodeler) [inline code/game/machinery/computer/card.dm:40]:
```suggestion
		"SolGov Representative",		// Wasp Edit - SolGov Rep
```

File: code/game/machinery/computer/crew.dm [hunk 13]
```diff
@@ -32,7 +32,7 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
 	var/list/jobs = new/list()
 	jobs["Captain"] = 00
 	jobs["Head of Personnel"] = 02
-	jobs["Lieutenant"] = 05
+	jobs["SolGov Representative"] = 05
```
Comment (Superyodeler) [inline code/game/machinery/computer/crew.dm:35]:
```suggestion
	jobs["SolGov Representative"] = 05		// Wasp Edit - SolGov Rep
```

File: code/game/objects/effects/landmarks.dm [hunk 14]
```diff
@@ -222,8 +222,8 @@ INITIALIZE_IMMEDIATE(/obj/effect/landmark)
 	icon_state = "Brig Physician"
 
 /obj/effect/landmark/start/lieutenant
-	name = "Lieutenant"
-	icon_state = "Lieutenant"
+	name = "SolGov Representative"
+	icon_state = "SolGov Representative"
```
Comment (Superyodeler) [inline code/game/objects/effects/landmarks.dm:226]:
```suggestion
	name = "SolGov Representative"		// Wasp Edit - SolGov Rep
	icon_state = "SolGov Representative"		// Wasp Edit - SolGov Rep
```
May also want to change the landmark from lieutenant to something else

File: code/game/objects/items/devices/PDA/PDA_types.dm [hunk 15]
```diff
@@ -197,10 +197,11 @@
...
 
-/obj/item/pda/lieutenant
-	name = "lieutenant PDA"
-	default_cartridge = /obj/item/cartridge/hos
-	icon_state = "pda-h"
+/obj/item/pda/solgov
+	name = "solgov officer PDA"
+	default_cartridge = /obj/item/cartridge/lawyer
+	inserted_item = /obj/item/pen/fountain/solgov
+	icon_state = "pda-solgov"
```
Comment (Superyodeler) [inline code/game/objects/items/devices/PDA/PDA_types.dm:204]: Needs modularizing

File: code/game/objects/items/devices/radio/headset.dm [hunk 16]
```diff
@@ -7,6 +7,7 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 	RADIO_CHANNEL_ENGINEERING = RADIO_TOKEN_ENGINEERING,
 	RADIO_CHANNEL_SECURITY = RADIO_TOKEN_SECURITY,
 	RADIO_CHANNEL_CENTCOM = RADIO_TOKEN_CENTCOM,
+	RADIO_CHANNEL_SOLGOV = RADIO_TOKEN_SOLGOV,
```
Comment (Superyodeler) [inline code/game/objects/items/devices/radio/headset.dm:10]:
```suggestion
	RADIO_CHANNEL_SOLGOV = RADIO_TOKEN_SOLGOV,		// Wasp Edit - SolGov Rep
```

File: code/game/objects/items/devices/radio/radio.dm [hunk 17]
```diff
@@ -239,7 +239,7 @@
 	var/datum/signal/subspace/vocal/signal = new(src, freq, speaker, language, message, spans, message_mods)
 
 	// Independent radios, on the CentCom frequency, reach all independent radios
-	if (independent && (freq == FREQ_CENTCOM || freq == FREQ_CTF_RED || freq == FREQ_CTF_BLUE))
+	if (independent && (freq == FREQ_CENTCOM || freq == FREQ_SOLGOV || freq == FREQ_CTF_RED || freq == FREQ_CTF_BLUE))
```
Comment (Superyodeler) [inline code/game/objects/items/devices/radio/radio.dm:242]:
```suggestion
	if (independent && (freq == FREQ_CENTCOM || freq == FREQ_SOLGOV || freq == FREQ_CTF_RED || freq == FREQ_CTF_BLUE))		// Wasp Edit - SolGov Rep
```

File: code/game/objects/items/storage/backpack.dm [hunk 18]
```diff
@@ -222,6 +222,7 @@
 /obj/item/storage/backpack/satchel/leather
 	name = "leather satchel"
 	desc = "It's a very fancy satchel made with fine leather."
+	icon = 'waspstation/icons/obj/storage.dmi'
```
Comment (Superyodeler) [inline code/game/objects/items/storage/backpack.dm:225]:
```suggestion
	icon = 'waspstation/icons/obj/storage.dmi'		// Wasp Edit - Suitcases
```

File: code/game/objects/items/storage/boxes.dm [hunk 19]
```diff
@@ -260,6 +260,7 @@
 
 /obj/item/storage/box/hypospray
 	name = "hypospray mk. II kit"
+	icon = 'waspstation/icons/obj/storage.dmi'
```
Comment (Superyodeler) [inline code/game/objects/items/storage/boxes.dm:263]:
```suggestion
	icon = 'waspstation/icons/obj/storage.dmi'		// Wasp Edit - Suitcases
```

File: code/game/say.dm [hunk 20]
```diff
@@ -14,6 +14,7 @@ GLOBAL_LIST_INIT(freqtospan, list(
 	"[FREQ_AI_PRIVATE]" = "aiprivradio",
 	"[FREQ_SYNDICATE]" = "syndradio",
 	"[FREQ_CENTCOM]" = "centcomradio",
+	"[FREQ_SOLGOV]" = "solgovradio",
```
Comment (Superyodeler) [inline code/game/say.dm:17]:
```suggestion
	"[FREQ_SOLGOV]" = "solgovradio",		// Wasp Edit - SolGov Rep
```

File: code/modules/admin/verbs/randomverbs.dm [hunk 21]
```diff
@@ -65,7 +65,7 @@
 		return
 
 	if (!sender)
-		sender = input("Who is the message from?", "Sender") as null|anything in list(RADIO_CHANNEL_CENTCOM,RADIO_CHANNEL_SYNDICATE)
+		sender = input("Who is the message from?", "Sender") as null|anything in list(RADIO_CHANNEL_CENTCOM,RADIO_CHANNEL_SYNDICATE,RADIO_CHANNEL_SOLGOV)
```
Comment (Superyodeler) [inline code/modules/admin/verbs/randomverbs.dm:68]:
```suggestion
		sender = input("Who is the message from?", "Sender") as null|anything in list(RADIO_CHANNEL_CENTCOM,RADIO_CHANNEL_SYNDICATE,RADIO_CHANNEL_SOLGOV)		// Wasp Edit - SolGov Rep
```

File: code/modules/admin/verbs/randomverbs.dm [hunk 22]
```diff
@@ -77,7 +77,7 @@
 
 	log_directed_talk(mob, H, input, LOG_ADMIN, "reply")
 	message_admins("[key_name_admin(src)] replied to [key_name_admin(H)]'s [sender] message with: \"[input]\"")
-	to_chat(H, "<span class='hear'>You hear something crackle in your ears for a moment before a voice speaks. \"Please stand by for a message from [sender == "Syndicate" ? "your benefactor" : "Central Command"]. Message as follows[sender == "Syndicate" ? ", agent." : ":"] <b>[input].</b> Message ends.\"</span>", confidential = TRUE)
+	to_chat(H, "<span class='hear'>You hear something crackle in your ears for a moment before a voice speaks. \"Please stand by for a message from [sender == RADIO_CHANNEL_SYNDICATE ? "your benefactor" : sender]. Message as follows[sender == "Syndicate" ? ", agent." : ":"] <b>[input].</b> Message ends.\"</span>", confidential = TRUE)
```
Comment (Superyodeler) [inline code/modules/admin/verbs/randomverbs.dm:80]:
```suggestion
	to_chat(H, "<span class='hear'>You hear something crackle in your ears for a moment before a voice speaks. \"Please stand by for a message from [sender == RADIO_CHANNEL_SYNDICATE ? "your benefactor" : sender]. Message as follows[sender == "Syndicate" ? ", agent." : ":"] <b>[input].</b> Message ends.\"</span>", confidential = TRUE)		// Wasp Edit - SolGov Rep
```

File: code/modules/jobs/access.dm [hunk 23]
```diff
@@ -178,7 +178,7 @@
 		if(6) //supply
 			return list(ACCESS_MAILSORTING, ACCESS_MINING, ACCESS_MINING_STATION, ACCESS_MECH_MINING, ACCESS_MINERAL_STOREROOM, ACCESS_CARGO, ACCESS_QM, ACCESS_VAULT)
 		if(7) //command
-			return list(ACCESS_HEADS, ACCESS_RC_ANNOUNCE, ACCESS_KEYCARD_AUTH, ACCESS_CHANGE_IDS, ACCESS_AI_UPLOAD, ACCESS_TELEPORTER, ACCESS_EVA, ACCESS_GATEWAY, ACCESS_ALL_PERSONAL_LOCKERS, ACCESS_HOP, ACCESS_CAPTAIN, ACCESS_VAULT, ACCESS_LIEUTENANT)
+			return list(ACCESS_HEADS, ACCESS_RC_ANNOUNCE, ACCESS_KEYCARD_AUTH, ACCESS_CHANGE_IDS, ACCESS_AI_UPLOAD, ACCESS_TELEPORTER, ACCESS_EVA, ACCESS_GATEWAY, ACCESS_ALL_PERSONAL_LOCKERS, ACCESS_HOP, ACCESS_CAPTAIN, ACCESS_VAULT, ACCESS_SOLGOV)
```
Comment (Superyodeler) [inline code/modules/jobs/access.dm:181]:
```suggestion
			return list(ACCESS_HEADS, ACCESS_RC_ANNOUNCE, ACCESS_KEYCARD_AUTH, ACCESS_CHANGE_IDS, ACCESS_AI_UPLOAD, ACCESS_TELEPORTER, ACCESS_EVA, ACCESS_GATEWAY, ACCESS_ALL_PERSONAL_LOCKERS, ACCESS_HOP, ACCESS_CAPTAIN, ACCESS_VAULT, ACCESS_SOLGOV)		// Wasp Edit - SolGov Rep
```

File: code/modules/jobs/access.dm [hunk 24]
```diff
@@ -337,8 +337,8 @@
 //WaspStation Begin
 		if(ACCESS_CLONING)
 			return "Cloning Room"
-		if(ACCESS_LIEUTENANT)
-			return "Lieutenant"
+		if(ACCESS_SOLGOV)
+			return "SolGov Office"
```
Comment (Superyodeler) [inline code/modules/jobs/access.dm:341]:
```suggestion
		if(ACCESS_SOLGOV)		// Wasp Edit - SolGov Rep
			return "SolGov Office"		// Wasp Edit - SolGov Rep
```

File: code/modules/jobs/access.dm [hunk 25]
```diff
@@ -384,6 +384,6 @@
 	if(jobName in get_all_centcom_jobs()) //Return with the NT logo if it is a CentCom job
 		return "CentCom"
 	for(var/datum/job/J in SSjob.occupations)
-		if((jobName in J.alt_titles) || (jobName == J.senior_title))
+		if((jobName in J.alt_titles) || (jobName == J.senior_title) && (J.title in get_all_job_icons()))
```
Comment (Superyodeler) [inline code/modules/jobs/access.dm:387]:
```suggestion
		if((jobName in J.alt_titles) || (jobName == J.senior_title) && (J.title in get_all_job_icons()))		// Wasp Edit - SolGov Rep
```

File: code/modules/jobs/jobs.dm [hunk 26]
```diff
@@ -37,7 +37,7 @@ GLOBAL_LIST_INIT(supply_positions, list(
 
 GLOBAL_LIST_INIT(service_positions, list(
 	"Head of Personnel",
-	"Lieutenant",
+	"SolGov Representative",
```
Comment (Superyodeler) [inline code/modules/jobs/jobs.dm:40]:
```suggestion
	"SolGov Representative",		// Wasp Edit - SolGov Rep
```

File: code/modules/mob/living/say.dm [hunk 27]
```diff
@@ -19,6 +19,7 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 	// Faction
 	RADIO_KEY_SYNDICATE = RADIO_CHANNEL_SYNDICATE,
 	RADIO_KEY_CENTCOM = RADIO_CHANNEL_CENTCOM,
+	RADIO_KEY_SOLGOV = RADIO_CHANNEL_SOLGOV,
```
Comment (Superyodeler) [inline code/modules/mob/living/say.dm:22]:
```suggestion
	RADIO_KEY_SOLGOV = RADIO_CHANNEL_SOLGOV,		// Wasp Edit - SolGov Rep
```

**Review by Superyodeler — DISMISSED** [review]
Sorry it took so long for me to get to this. Everything seems to work fine, just needs modularizing.

**Review by Maldaris — APPROVED** [review]

**Pull request merged**